src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -597,9 +599,17 @@ private SubpatternSyntax ParseSubpatternElement()
             if (this.CurrentToken.Kind == SyntaxKind.ColonToken && ConvertPatternToExpressionIfPossible(pattern, permitTypeArguments: true) is ExpressionSyntax expr)
             {
                 var colon = EatToken();
-                exprColon = expr is IdentifierNameSyntax identifierName
-                    ? _syntaxFactory.NameColon(identifierName, colon)
-                    : _syntaxFactory.ExpressionColon(CheckFeatureAvailability(expr, MessageID.IDS_FeatureExtendedPropertyPatterns), colon);
+                if (expr is IdentifierNameSyntax identifierName)
+                {
+                    exprColon = _syntaxFactory.NameColon(identifierName, colon);
+                }
+                else
+                {
+                    expr = allowExtendedProperties
+                        ? CheckFeatureAvailability(expr, MessageID.IDS_FeatureExtendedPropertyPatterns)
+                        : AddError(expr, ErrorCode.ERR_IdentifierExpected);
```

Comment:
Instead of this I think it would be simpler to just check `ExpressionColon.Expression is IdentifierName` at use-sites and report. That would make this a semantic error but we're already checking invalid names there.